pkg/api/vm/handler.go

```diff
@@ -562,6 +572,43 @@ func (h *vmActionHandler) copySecret(sourceName, targetName string, templateVers
 
 }
 
+func (h *vmActionHandler) createVmImages(templateVersion *harvesterv1.VirtualMachineTemplateVersion, vm *kubevirtv1.VirtualMachine) error {
+	for index, volume := range vm.Spec.Template.Spec.Volumes {
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+		vmImageName := getTemplateVersionVmImageName(templateVersion.Name, index)
+		vmImage := &harvesterv1.VirtualMachineImage{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      vmImageName,
+				Namespace: vm.Namespace,
+				Annotations: map[string]string{
+					util.AnnotationStorageClassName: "harvester-longhorn",
+				},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: templateVersion.APIVersion,
+						Kind:       templateVersion.Kind,
+						Name:       templateVersion.Name,
+						UID:        templateVersion.UID,
+					},
+				},
+			},
+			Spec: harvesterv1.VirtualMachineImageSpec{
+				DisplayName:  vmImageName,
+				SourceType:   harvesterv1.VirtualMachineImageSourceTypeExportVolume,
+				PVCName:      volume.PersistentVolumeClaim.ClaimName,
+				PVCNamespace: vm.Namespace,
+			},
+		}
+
+		if _, err := h.vmImages.Create(vmImage); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
is the plan to only use the default storage class or should we be copying the storage class from the underlying pvc?

Reply:
Do you mean `storageClassParameters` in VMImage?

Reply:
yes.. should we copy these?

Reply:
This is a good question. I prefer not to copy it because we don't let users modify `storageClassParameters` from the dashboard or terraform. It's better to let the webhook mutator patch the default setting currently.

Reply:
in the current form new VM's launched will be on the default storage class

Reply:
Hi @ibrokethecloud, I already fixed this part. Could you review it again? Thank you.